sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/ServerCertificateCheck.kt

```diff
@@ -104,6 +125,43 @@ class ServerCertificateCheck : AbstractCheck() {
         }
     }
 
+    private class AndroidWebViewCandidateVisitor : KtVisitorVoid() {
+        private val androidSslErrorHandlerProceedFunMatcher = FunMatcher {
+            definingSupertype = "android.webkit.SslErrorHandler"
+            withNames("proceed")
+        }
+
+        private val androidSslErrorHandlerCancelFunMatcher = FunMatcher {
+            definingSupertype = "android.webkit.SslErrorHandler"
+            withNames("cancel")
+        }
+
+        var foundProceedCall: Boolean = false
+        var foundCancelCall: Boolean = false
+        var potentialBranching: Boolean = false
+
+        override fun visitCallExpression(expression: KtCallExpression) {
+            val isProceedCall = androidSslErrorHandlerProceedFunMatcher.matches(expression)
+            val isCancelCall = androidSslErrorHandlerCancelFunMatcher.matches(expression)
+            foundProceedCall = foundProceedCall || isProceedCall
+            foundCancelCall = foundCancelCall || isCancelCall
```

Comment:
here we always try to match, both proceed and cancel and always override the foundProceedCall and foundCancelCall.

It seems to me we can inline 
```suggestion
            foundProceedCall = foundProceedCall || androidSslErrorHandlerProceedFunMatcher.matches(expression)
            foundCancelCall = foundCancelCall || androidSslErrorHandlerCancelFunMatcher.matches(expression)
```

Reply:
Also since these calls can't be matched together, we can refactor to only match one. I'll let you decide how is better to refactor it.